src/new_hope/insertionsort1.clj

```diff
@@ -0,0 +1,49 @@
+(ns new-hope.insertionsort1
+  (:require [midje.sweet :refer :all]))
+
+; https://www.hackerrank.com/challenges/insertionsort1
+
+(def arr [2 4 6 8 3])
+
+(def arr-size (count arr))
+(def elem (last arr))
+
+
+(defn insert-elem [arr]
```

Comment:
This looks really hairy, sorry :) And also does not work with lists: `(insert-elem '(5 6 9 7))`
I think that insertion sort could traverse the input once and build a list as a result, it supports insertions in a more natural way than any other data structure.